nav2_costmap_2d/src/footprint.cpp

```diff
@@ -210,14 +210,18 @@ bool makeFootprintFromString(
   return true;
 }
 
-std::vector<geometry_msgs::msg::Point> makeFootprintFromParams(rclcpp::Node::SharedPtr node)
+std::vector<geometry_msgs::msg::Point> makeFootprintFromParams(
+  nav2_lifecycle::LifecycleNode::SharedPtr node)
 {
   std::string full_param_name = "footprint";
   std::string full_radius_param_name = "robot_radius";
   std::vector<geometry_msgs::msg::Point> points;
   std::string footprint;
 
-  node->get_parameter_or<std::string>(full_param_name, footprint, "[]");
+  node->declare_parameter(full_param_name, rclcpp::ParameterValue("[]"));
```

Comment:
If this function is ever called more than once, an error on re-declaring the parameter would occur, right?

Reply:
@bpwilcox I've scanned the code and this function is not called. I'd prefer to remove it if we're not using it. The parent node, costmap_2d_ros, already has a "footprint_" string parameter that is then passed to makeFootprintFromString. What do you think?

Reply:
I'd agree we probably don't need this function

Reply:
I'd agree we probably don't need this function

Reply:
Ok, removed.